### PR TITLE
Package-ize common

### PR DESCRIPTION
**Pull request by mrice32**
This adds the `@umaprotocol/common` package. For simplicity, all elements that were previously exported by individual files in common are just exported by the `index.js` file. This can be reorganized later if desired.

Notes:
- Certain files (like the truffle config file) were expected to be imported as `const object = require...`. These were all changed to `const { object } = require...` to refrain from polluting the exports of the `index.js` file.
- The abi-related files in common required some refactoring to begin importing the contracts via the `core` package. We should ensure that the imports still work with webpack when used in the voter dapp to explain admin requests.
- The voter-dapp installation in ci was failing because it was still attempting to install the dependencies via npm.
- Some dependencies needed to be added to the voter-dapp after the above change. I'm not exactly sure why the build began failing _now_ (after the move to yarn/lerna), but adding dependencies seemed to fix it. There are some new warnings in the voter-dapp, so we should be wary of releasing it until it's been fully tested.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -136,18 +136,6 @@ jobs:
           name: Deploy Contracts
           working_directory: ~/protocol/core
           command: $(npm bin)/truffle migrate --reset --network test
-      - restore_cache:
```
Comment (chrismaree) [inline .circleci/config.yml:139]: Just to understand why did you change this?

File: common/index.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
+module.exports = {
```
Comment (chrismaree) [inline common/index.js:1]: pitty this is required but it's not too bad all things considered.

File: core/scripts/umip-3/4_ExecuteSimulate.js [hunk 3]
```diff
@@ -3,10 +3,15 @@
 
 const assert = require("assert").strict;
 
-const { getRandomUnsignedInt } = require("../../../common/Random.js");
-const { advanceBlockAndSetTime, takeSnapshot, revertToSnapshot } = require("../../../common/SolidityTestUtils.js");
-const { RegistryRolesEnum } = require("../../../common/Enums.js");
-const { computeVoteHash } = require("../../../common/EncryptionHelper.js");
+const {
```
Comment (chrismaree) [inline core/scripts/umip-3/4_ExecuteSimulate.js:6]: These imports are much nicer.

File: .circleci/config.yml (hunk 1 above)
Comment (mrice32) [reply .circleci/config.yml:139]: Yeah, this should've been changed before, when we moved the voter-dapp to lerna. It should be relying on the `npx lerna bootstrap` called in the `checkout_and_install` step to install all of its dependencies. However, the local continued to work until it needed to install other sub-packages (like common).

File: common/AbiUtils.js [hunk 4]
```diff
@@ -23,14 +23,15 @@ function getAllContracts() {
 
     // Note: all arguments must be hardcoded here for webpack to bundle the files correctly.
     // This line also generates a few build warnings that should be ignored.
-    const contractContext = require.context("../core/build/contracts/", true, /\.json$/);
+    const contractContext = require.context("@umaprotocol/core/build/contracts/", true, /\.json$/);
```
Comment (mrice32) [inline common/AbiUtils.js:26]: This might break the voter-dapp. I'm going to test.

File: common/index.js (hunk 2 above)
Comment (mrice32) [reply common/index.js:1]: Yeah, it could be worthwhile to have this auto-detect files and add them automatically, but I know some tools have trouble with dynamic requires (like webpack, I think).

**Comment by mrice32** [comment]
> I think the refactor in the `/core` directory is going to be pretty difficult

@chrismaree  Yeah, there are definitely a few tweaks we'll need to make there. The rough plan is to make sure that the `core/` exports work for truffle artifacts (after the truffle release today). Then we can move the bots to running from their own directories and move the CLI tool out of core. Then we should discuss what other types of exports we might want from core. Pretty excited to move this refactor along -- I think things will feel much easier and cleaner.

File: common/AbiUtils.js (hunk 4 above)
Comment (mrice32) [reply common/AbiUtils.js:26]: Something _else_ in common broke the voter-dapp. It's now fixed.
